parse/Expression.ts

```diff
@@ -0,0 +1,56 @@
+import { Error } from "@cogneco/mend"
+import { Expression } from "../Expression"
+import * as lexer from "../lexer"
+import { Value } from "../Value"
+import { Source } from "./Source"
+
+export function parseExpression(source: Source): Expression
+export function parseExpression(source: string, handler?: Error.Handler): Expression
+export function parseExpression(source: string | Source, handler?: Error.Handler): Expression {
+	if (typeof source == "string") {
+		handler = handler || new Error.ConsoleHandler()
+		const tokens = lexer.tokenize(source, handler).toArray()
+		source = new Source(tokens, handler)
+	}
+	handler = handler instanceof Expression ? handler : undefined
+	let result: Expression | undefined
+	let algebra = false
+	let comparator = false
+	while (+!result ^ +algebra) {
```

Comment:
`!result != algebra`

(or `!result != !!algebra` but algebra is already a boolean)